core/shared/src/main/scala/fs2/async/mutable/Queue.scala

```diff
@@ -199,7 +199,7 @@ object Queue {
                   F.guaranteeCase(d.get) {
                     case ExitCase.Completed => F.unit
                     case ExitCase.Error(t)  => cleanup *> F.raiseError(t)
-                    case ExitCase.Canceled  => cleanup *> F.never
+                    case ExitCase.Canceled  => cleanup *> F.unit
                   }
```

Comment:
I think this is a remnant of the old `F` cancellation model where you'd make canceled computations non-terminating and use `onCancelRaiseError`. That would explain why it regressed in `M4`. The fix makes sense to me intuitively.

Reply:
Yep, that's a good way to think about it. Makes sense now in hindsight.